### DISCUSSION TITLE
Reworked interface and UX

### DISCUSSION THREAD

**Discussion by theGreatWhiteShark** [hydrogen-music/hydrogen#1402] [General]
Hey everyone,

I'm very happy to announce that Hydrogen has now a reworked and more modern UI!

![reworked_gui](https://user-images.githubusercontent.com/14164547/141465508-3790f930-8682-42f6-878c-25240fbe97f2.png)

The main new features are:

- Improved scalability (most PNG images were replaced by SVGs, hardcoded PNG labels are now directly drawn by Qt, and spin boxes, buttons, and combo boxes are now based on native Qt widgets)
- Improved internationalization (all labels are translatable now and support UTF-8)
- Improved accessibility (widgets were increased to fill available space, preference option to (de- and) increase font size)
- Input widgets (rotary, fader, combo box, spin box, button) do support both mouse and keyboard input (e.g. setting numerical values)
- Input widgets are faintly highlighted when hovered and more strongly highlighted when clicked (focuses). Only the focused widget can receive keyboard input
- MIDI-learnable widgets now show their corresponding MIDI action and it's binding in the tooltip
- All colors can be altered via the preferences
- All options set in the Appearance tab (including all colors of the UI) can be imported and exported into and shared via dedicated .h2theme files.

The code is still hot and already merged into master (no bundled release yet!). I would be very glad if you could give it a shot and help to get it nice and stable before we release it via the next beta version of Hydrogen.

I also want to encourage you to toy with the color settings in Preferences and to share your custom themes with the community. The best ones will be bundled with the upcoming 1.2 release! (My focus was on improving the UX and not so much on beautifying the UI. So, there is much of room for improvement.)

**Comment by mahula** [comment]
Great to hear!
Great work - thank you very much.
Will this be part of a next release?

**Reply by theGreatWhiteShark** [reply]
> Will this be part of a next release?

Sure.

**Comment by oddtime** [comment]
After you push the buttons to increase/lower the bpm, then the space key it is not working as play/stop function (but it enters the character in the widget textline).
I would expect the space key acting as play/stop unless you don't click inside the bpm widget texline.
What do you think?

**Reply by theGreatWhiteShark** [reply]
Yes. This one is breaking the UX found in previous Hydrogen releases. But unfortunately there is not much we can do about it. The buttons are part of the `QSpinBox` and we can either use this well tested and established widget our create a custom one with a QLineEdit and two buttons like the one we had beforehand. But this would mean to implement/replicate/maintain all the logic of the Qt spin box ourselves. On the other hand, the latter does not allow for such a fine-grained control the previous UX could be kept. Once the user clicks within the boundaries of the widget, it will have focus and the only thing we could do to make the space key toggle playback instead of inserting a space in the text edit. But this would feel highly awkward.
